In [2]:
import os
import numpy as np
import pylab as pl
import json
from scipy.optimize import curve_fit,minimize
c=json.load(open('pui.json'))
pl.rcParams.update(c)

%pylab inline



Populating the interactive namespace from numpy and matplotlib


In [3]:
alpha = 0.05
P_0= 27.2*0.01
P_1=28.2*0.01
n_0=409
n_1=564
nt_0=P_0*n_0
nt_1=P_1*n_1


In [4]:
sp=(P_0*n_0+P_1*n_1)/(n_1+n_0)
print sp

0.277796505653


In [5]:
sp_stdev= lambda p, n: np.sqrt(p*(1-p)/n)

sp_stdev_2y=sp_stdev(P_0,n_0+n_1)
print P_0, n_0, n_1, sp_stdev_2y

0.272 409 564 0.0142657208499


In [7]:
zscore = lambda p0, p1, s : (p0-p1)/s
z_2y = zscore(P_1, P_0, sp_stdev_2y)
print z_2y


0.700981051377


In [9]:
p_2y=1-0.7734
def report_result(p,a):
    print 'is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a)
    if p<a:
        print "YES!"
    else: print "NO!"
    
    print 'the Null hypothesis is {}'.format( 'rejected' if p<a  else 'not rejected') 

    
report_result(p_2y,alpha)


is the p value 0.23 smaller than the critical value 0.05? 
NO!
the Null hypothesis is not rejected


In [10]:
Ntot = 973
expected = 564*409*131.579*841.421
sample_values = [[14.7*5.64,85.3*5.64],[11.9*4.09,88.1*4.09]]
 
chisqstat= lambda N, values, expect : N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect)

print chisqstat(Ntot,  sample_values, expected)

1.58939333193


we have 1 degree of freedom and 0.05 is alpha so intersecting their respective row and column we get 3.84. 1.589 is close to 3.84. So we cannot reject the null hypothesis . AS the p-value for the corresponding value is not below 0.05. (can see from the table of 'Percentage points of Chi-Square distribution')